### PR TITLE
MDEV-18869 Assertion `!((field)->vcol_info && (field)->stored_in_db()…

### PR DESCRIPTION
**Pull request by midenok**
This code is submitted under the BSD-new license.

### CODE REVIEW

File: mysql-test/suite/versioning/t/alter.test [hunk 1]
```diff
@@ -469,5 +469,12 @@ alter table t add check (a > 0);
 insert into t values (0);
 insert into t values (2);
 
+--echo #
+--echo # MDEV-18869 Assertion `!((field)->vcol_info && (field)->stored_in_db())' failed in innodb_col_no upon altering table with system versioning
+--echo #
+create or replace table t1 (a int, b int generated always as (0) stored) engine=innodb with system versioning;
+set system_versioning_alter_history= keep;
+alter table t1 modify a int with system versioning;
```
Comment (dr-m) [inline mysql-test/suite/versioning/t/alter.test]: Please test also with a `generated…virtual` column, and with `ALGORITHM=COPY`.

I would prefer to have some data inside the table, and some `SELECT` to demonstrate that the operation not only does not crash, but actually is producing the correct results.

If you have some data inside the table, then
```sql
--enable_info
ALTER TABLE …;
--disable_info
```
will show "0 rows affected" if native ALTER was used, and a nonzero count if `ALGORITHM=COPY` was used.

File: storage/innobase/handler/handler0alter.cc [hunk 2]
```diff
@@ -8587,7 +8587,7 @@ vers_change_fields_cache(
 	    ha_alter_info->alter_info->create_list);
 
 	while (const Create_field* create_field = it++) {
-		if (!create_field->field) {
+		if (!create_field->field || create_field->field->vcol_info) {
```
[the remark this answers is not in the document]
Comment (midenok) [reply storage/innobase/handler/handler0alter.cc]: Answered [here](https://jira.mariadb.org/browse/MDEV-18869?focusedCommentId=125242&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-125242)

File: mysql-test/suite/versioning/t/alter.test (hunk 1 above)
Comment (midenok) [reply mysql-test/suite/versioning/t/alter.test]: Fixed.

**Comment by dr-m** [comment]
Applied in a138d061b5fe2e211f80ec3f6080316c41c04f07.

**Pull request closed without merging**